File: tp/61236 - Drachenberg, Franco Eduardo/tp6/backend/app/schemas/carrito.py
from pydantic import BaseModel, Field


class ItemCarritoCreate(BaseModel):
    producto_id: int
    cantidad: int = Field(ge=1)


class CheckoutRequest(BaseModel):
    direccion: str = Field(min_length=1, max_length=255)
    tarjeta: str = Field(min_length=12, max_length=32)


class ItemCarritoRead(BaseModel):
    producto_id: int
    nombre: str
    precio: float
    cantidad: int
    subtotal: float
    imagen: str | None = None


class CarritoRead(BaseModel):
    id: int
    estado: str
    total: float
    items: list[ItemCarritoRead]


class CheckoutResponse(BaseModel):
    compra_id: int
    subtotal: float
    iva: float
    envio: float
    total: float

File: tp/61719 - Amoedo, Pablo Isaías/tp6/frontend/app/services/carrito.ts
'use client';

import { api } from '@/lib/api';

export interface ItemCarrito {
  carrito_id: number;
  producto_id: number;
  nombre: string;
  imagen: string;
  precio_unitario: number;
  cantidad: number;
  subtotal: number;
  iva: number;
}

export interface Carrito {
  carrito: ItemCarrito[];
  subtotal: number;
  iva: number;
  envio: number;
  total: number;
}

export async function obtenerCarrito(): Promise<Carrito> {
  const response = await api.get('/carrito');
  return response.data;
}

export async function agregarAlCarrito(producto_id: number, cantidad: number = 1): Promise<any> {
  const response = await api.post('/carrito', { producto_id, cantidad });
  return response.data;
}



export async function eliminarDelCarrito(producto_id: number): Promise<any> {
  const response = await api.delete(`/carrito/${producto_id}`);
  return response.data;
}

export async function cancelarCarrito(): Promise<any> {
  const response = await api.delete('/carrito');
  return response.data;
}

export interface CompraRequest {
  direccion: string;
  tarjeta: string;
}

export async function finalizarCompra(compraData: CompraRequest): Promise<any> {
  const response = await api.post('/comprar', compraData);
  return response.data;
}
